src/leb128.rs

```diff
@@ -0,0 +1,512 @@
+//! Read and write DWARF's "Little Endian Base 128" (LEB128) variable length
+//! integer encoding.
+//!
+//! The implementation is a direct translation of the psuedocode in the DWARF 4
+//! standard's appendix C.
+//!
+//! Read and write signed integers:
+//!
+//! ```
+//! use gimli::leb128;
+//!
+//! let mut buf = [0; 1024];
+//!
+//! // Write to anything that implements `std::io::Write`.
+//! {
+//!     let mut writable = &mut buf[..];
+//!     leb128::write::signed(&mut writable, -12345).expect("Should write number");
+//! }
+//!
+//! // Read from anything that implements `std::io::Read`.
+//! let mut readable = &buf[..];
+//! let val = leb128::read::signed(&mut readable).expect("Should read number");
+//! assert_eq!(val, -12345);
+//! ```
+//!
+//! Or read and write unsigned integers:
+//!
+//! ```
+//! use gimli::leb128;
+//!
+//! let mut buf = [0; 1024];
+//!
+//! {
+//!     let mut writable = &mut buf[..];
+//!     leb128::write::unsigned(&mut writable, 98765).expect("Should write number");
+//! }
+//!
+//! let mut readable = &buf[..];
+//! let val = leb128::read::unsigned(&mut readable).expect("Should read number");
+//! assert_eq!(val, 98765);
+//! ```
+
+use std;
+
+#[doc(hidden)]
+pub const CONTINUATION_BIT: u8 = 1 << 7;
+#[doc(hidden)]
+pub const SIGN_BIT: u8 = 1 << 6;
+
+#[doc(hidden)]
+#[inline]
+pub fn low_bits_of_byte(byte: u8) -> u8 {
+    byte & !CONTINUATION_BIT
+}
+
+#[doc(hidden)]
+#[inline]
+pub fn low_bits_of_u64(val: u64) -> u8 {
```

Comment:
All of these `doc(hidden)` things can be `pub(crate)` now, instead. Obviously doesn't need to block anything from landing.